File: server/NotificationService/NotificationService.Application/Events/OrderCancelledEvent.cs


namespace NotificationService.Application.Events
{
    public record OrderCancelledEvent(
        Guid OrderId,
        Guid CustomerId,
        List<OrderCancelledItemEvent> Items,
        DateTime CancelledAt
    );

    public record OrderCancelledItemEvent(
        Guid ProductId,
        string ProductName,
        int Quantity
    );
}

File: server/OrderService/order-service/src/main/java/com/nutrihub/orderservice/infrastructure/messaging/events/OrderCancelledEvent.java
package com.nutrihub.orderservice.infrastructure.messaging.events;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

public record OrderCancelledEvent(
        UUID orderId,
        UUID customerId,
        List<OrderItemEvent> items,
        LocalDateTime cancelledAt
) implements Serializable {
    public record OrderItemEvent(
            UUID productId,
            String productName,
            int quantity
    ) implements Serializable {}
}
